x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/SqlParserTests.java

```diff
@@ -254,6 +261,40 @@ public void testLimitToPreventStackOverflowFromLargeComplexSubselectTree() {
             e.getMessage());
     }
 
+    public void testLimitStackOverflowForInAndLiteralsIsNotApplied() {
+        new SqlParser().createStatement("SELECT * FROM t WHERE a IN(" +
```

Comment:
Only the parser being invoked is enough here? Can you check that at least that a minimal plan is created correctly given the query? Also, there is already a method that creates the statement given a query String: `parseStatement(String)`.

Reply:
this test is only to check that we don't hit the limit, but I can add more assertions.